### PR TITLE
Adding support for fallback domains

### PR DESCRIPTION
**Pull request by kjhandy**

### CODE REVIEW

**Review by patryk — CHANGES_REQUESTED** [review]
Missing documentation, tests failing, improper cloudflare-go fields usage.

**Comment by kjhandy** [comment]
Added documentation. Tests should all pass once cloudflare-go is bumped (depends on https://github.com/cloudflare/cloudflare-go/pull/761).

**Comment by kjhandy** [comment]
@patryk I _believe_ this is waiting on a cloudflare-go release and to bump the version here. Is there a plan to do that?

File: cloudflare/resource_cloudflare_fallback_domain_test.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	})
+}
+
+func testAccCloudflareFallbackDomain(rnd, accountID string, description string, suffix string, dns_server []string) string {
+	return fmt.Sprintf(`
+resource "cloudflare_fallback_domain" "%[1]s" {
+  account_id = "%[2]s"
+  domains {
+    description = "%[3]s"
+    host = "%[4]s"
```
Comment (jacobbednarz) [inline cloudflare/resource_cloudflare_fallback_domain_test.go:59]: should this be `suffix`?

```
=== RUN   TestAccCloudflareFallbackDomain
    resource_cloudflare_fallback_domain_test.go:25: Step 1/2 error: Error running pre-apply refresh: exit status 1

        Error: Unsupported argument

          on terraform_plugin_test.tf line 6, in resource "cloudflare_fallback_domain" "hjygxhxvvw":
           6:     host = "example.com"

        An argument named "host" is not expected here.
```

File: cloudflare/resource_cloudflare_fallback_domain_test.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudflareFallbackDomain(rnd, accountID, "example domain", "example.com", []string{"2.2.2.2"}),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(name, "account_id", accountID),
+					resource.TestCheckResourceAttr(name, "domains.0.description", "example domain"),
+					resource.TestCheckResourceAttr(name, "domains.0.suffix", "example.com"),
+					resource.TestCheckResourceAttr(name, "domains.0.dns_server", "['2.2.2.2']"),
```
Comment (jacobbednarz) [inline cloudflare/resource_cloudflare_fallback_domain_test.go:37]: i think you want the first entry in the slice here

```suggestion
					resource.TestCheckResourceAttr(name, "domains.0.dns_server.0", "2.2.2.2"),
```

File: cloudflare/resource_cloudflare_fallback_domain_test.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+				),
+			},
+			{
+				Config: testAccCloudflareFallbackDomain(rnd, accountID, "second example domain", "example_two.com", []string{"1.1.1.1"}),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(name, "account_id", accountID),
+					resource.TestCheckResourceAttr(name, "domains.0.description", "second example domain"),
+					resource.TestCheckResourceAttr(name, "domains.0.host", "example_two.com"),
+					resource.TestCheckResourceAttr(name, "domains.0.dns_server", "['1.1.1.1']"),
```
Comment (jacobbednarz) [inline cloudflare/resource_cloudflare_fallback_domain_test.go:46]: same as above, i think you want the first entry.

```suggestion
					resource.TestCheckResourceAttr(name, "domains.0.dns_server.0", "1.1.1.1"),
```

File: website/docs/r/fallback_domain.html.markdown [hunk 4]
```diff
@@ -0,0 +1,47 @@
+---
+layout: "cloudflare"
+page_title: "Cloudflare: cloudflare_fallback_domain"
+sidebar_current: "docs-cloudflare-resource-fallback-domain"
```
Comment (jacobbednarz) [inline website/docs/r/fallback_domain.html.markdown:4]: don't forget to add a link this to document in https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/website/cloudflare.erb so it appears in the sidebar.

**Pull request merged**